Lib/sqlite3/dump.py

```diff
@@ -73,9 +78,14 @@ def _iterdump(connection):
                 "||quote({0})||".format(_quote_name(col)) for col in column_names
             )
         )
-        query_res = cu.execute(q)
-        for row in query_res:
-            yield("{0};".format(row[0]))
+        orig_text_factory = connection.text_factory
+        try:
+            connection.text_factory = bytes
+            query_res = cu.execute(q)
+            for row in query_res:
+                yield("{0};".format(_force_decode(row[0])))
```

Comment:
Is it safe to assume default bytes.decode() arguments here (UTF-8) or does sqlite store the DB encoding somewhere?

Reply:
Offset 0x38 in the header stores the encoding. https://www.sqlite.org/fileformat.html#the_database_header

By default it appears to be UTF-8. But it could also be UTF-16le or UTF-16be.

Here's how to create a db with alternate encodings. https://www.sqlite.org/pragma.html#pragma_encoding
```
PRAGMA encoding='UTF-16';
BEGIN TRANSACTION;
CREATE TABLE foo (data VARCHAR);
INSERT INTO foo VALUES (CAST(X'619f' AS VARCHAR));
COMMIT;
```

Reply:
But perusing here https://github.com/search?q=repo:python/cpython+path:/^Modules\/_sqlite/+utf&type=code makes me think Python only supports UTF-8.

Reply:
That was my conclusion as-well, it looks like c portion of the sqlite module also makes the assumption https://github.com/python/cpython/blob/main/Modules/_sqlite/cursor.c#L400C53-L400C53 .

Reply:
The sqlite3 extension only supports UTF-8, and there is no plan to change that; it would increase the complexity of both the extension and the test suite for little gain (there has been no feature requests to change this).

Reply:
I'm sure most anyone who have had to deal with Win32's widechar mess probably considers UTF-16 more of a bug than a feature anyways ;) Just was not sure if I was missing something with the decode.